src/Middleware/UserPageDecorationMiddleware.php

```diff
@@ -0,0 +1,193 @@
+        if (method_exists($page, 'getControlPanel')) {
+            $tpl_data['control_panel'] = $page->getControlPanel();
+        }
+        if (method_exists($page, 'getFeedbackPanel')
+            && $user->hasPermission('bvl_feedback')
+        ) {
+            $tpl_data['feedback_panel'] = $page->getFeedbackPanel(
+                $get['candID'],
+                $get['sessionID'] ?? null
+            );
+
+            $tpl_data['bvl_feedback'] = NDB_BVL_Feedback::bvlFeedbackPossible(
+                $this->page
+            );
+        }
+
+        // This shouldn't exist. (And if it does, it shouldn't reference
+        // mantis..)
+        $tpl_data['issue_tracker_url'] = $this->Config->getSetting('issue_tracker_url');
+
+        // We're back in the territory of stuff that belongs here..
+
+        // Variables that get passed along to the LorisHelper javascript object.
+        $tpl_data['studyParams'] = array(
+                                    'useEDC'      => $this->Config->getSetting('useEDC'),
+                                    'useProband'  => $this->Config->getSetting(
+                                        'useProband'
+                                    ),
+                                    'useFamilyID' => $this->Config->getSetting(
+                                        'useFamilyID'
+                                    ),
+                                    'useConsent'  => $this->Config->getSetting(
+                                        'ConsentModule'
+                                    )['useConsent'],
+                                   );
+        $tpl_data['jsonParams']  = json_encode(
+            array(
+             'BaseURL'   => $this->BaseURL,
+             'TestName'  => $tpl_data['test_name'] ?? '',
+             'Subtest'   => $tpl_data['subtest'] ?? '',
+             'CandID'    => $tpl_data['candID'] ?? '',
+             'SessionID' => $tpl_data['sessionID'] ?? '',
+             'CommentID' => $tpl_data['commentID'] ?? '',
+            )
+        );
+
+        // User related template variables that used to be in main.php.
+        $site_arr = $this->user->getData('CenterIDs');
+        foreach ($site_arr as $key=>$val) {
+            $site[$key]        = & \Site::singleton($val);
+            $isStudySite[$key] = $site[$key]->isStudySite();
+        }
+
+        $oneIsStudySite   = in_array("1", $isStudySite);
+        $tpl_data['user'] = $this->user->getData();
+        $tpl_data['user']['permissions']          = $this->user->getPermissions();
+        $tpl_data['user']['user_from_study_site'] = $oneIsStudySite;
+        $tpl_data['userNumSites']         = count($site_arr);
+        $tpl_data['user']['SitesTooltip'] = str_replace(
+            ";",
+            "<br/>",
+            $this->user->getData('Sites')
+        );
+
+        $tpl_data['hasHelpEditPermission'] = $this->user->hasPermission(
+            'context_help'
+        );
+
+        // FIXME: This should be array_filter
```

Comment:
```
$perms = $this->user->getPermissions();
$realPerms = array_reduce(array_keys($perms), function($acc, $p) use ($perms) {
    if ($perms[$p] == 1) {
        $acc[] = $k;
    }
    return $acc;
}, []);
```

Reply:
not sure about `$perms[$p] == 1`

Reply:
That seems less readable. If anything, it should be hidden behind a function in the User class and not part of this PR, otherwise I doubt code like that will be maintained.